dwave/optimization/_model.pyx

```diff
@@ -1359,6 +1359,12 @@ def _split_indices(indices):
     return tuple(basic_indices), tuple(advanced_indices)
 
 
+def as_array_symbol(model, symbol):
+    if isinstance(symbol, ArraySymbol):
+        return symbol
+    return model.constant(np.asarray(symbol, dtype=np.double))
```

Comment:
```python
    try:
        return model.constant(symbol)
    except TypeError as err:
        raise TypeError(f"{symbol!r} cannot be interpreted as an array symbol") from err
```
might be a bit more explicit for the context we want to use it and it also let's `Constant` do the work. We should triple check that `Constant` raises a `TypeError` when we expect it to.

Reply:
Agreed!